### ISSUE TITLE
Safety in parallel builds

### ISSUE THREAD

**Issue by hainest** [Daltz333/sphinxcontrib-remoteliteralinclude#21]
dcb06b02 marked rli safe for parallel reading, but not for parallel writing. I couldn't find any comments or documentation describing why that was chosen. I'm definitely not a sphinx expert, but a cursory look over the code it's not immediately clear to me where thread unsafe writing would occur. Anecdotally, enabling parallel writes in my build with a few hundred files and a couple dozen rli uses, I didn't see any problems.

**Comment by joshuacwnewton** [comment]
Hullo! Arriving here because I ran into the following warning when building my docs with this extension on RTD:

```
WARNING: the sphinxext.remoteliteralinclude extension is not safe for parallel writing
WARNING: doing serial write
```

Since I have `fail_on_warning: true` set in my `.readthedocs.yaml` config file, the build fails.

> A secondary cause here is the fact that ReadTheDocs uses parallel writing by default, e.g. the Sphinx command:
> 
> ```
>  	python -m sphinx -T -W --keep-going -j auto -b html -d _build/doctrees -D language=en . $READTHEDOCS_OUTPUT/html
> ```
>
> `-j auto` is the issue here, since it parallelizes by default across all available threads.

I think to get around this (when combining this extension with ReadTheDocs and `fail_on_warning=True`), I'm going to need to use a custom build step in my RTD settings that omits `-j auto`, which is a bit of a pain but not unworkable.

**Comment by hainest** [comment]
@joshuacwnewton Yeah, I ended up just running `sed` on `remoteliteralinclude.py` to replace `"parallel_write_safe": False` with `"parallel_write_safe": True`. Definitely hackish, but I never got a response here. :shrug:

**Comment by joshuacwnewton** [comment]
@sciencewhiz seems to have been very active ~2 days ago, maybe a quick ping could help us out.

Even a config option for `conf.py` akin to `allow_parallel_writes = True` could be helpful, assuming there really is some sort of risk there that the maintainers are trying to avoid. Maybe that would be a bit hacky (given that it would obscure that the warning/fallback to sequential is coming from Sphinx, not the extension) but it would at least let power users flip a switch if they're willing to take the risk.
